internal/controller/children.go

```diff
@@ -101,14 +102,34 @@ func (c ChildResource[T]) Reconcile(ctx context.Context, clnt client.Client, cr
 			if !ok {
 				return fmt.Errorf("Could not cast when reconciling")
 			}
+			events := v1.EventList{}
+			selector, err := fields.ParseSelector(fmt.Sprintf("involvedObject.name=%s,involvedObject.namespace=%s,involvedObject.kind=StatefulSet", current.ObjectMeta.Name, current.ObjectMeta.Namespace))
+			if err == nil {
```

Comment:
If you reverse some check you could have much less nested code, e.g.:
```
if err != nil {
  log.Error(...)
}
if err := clnt.List[....]
```

Reply:
I on purpose have else statements because I don't error out on errors but want to continue normally. If for some reason we can't get events, the other code still works. I mean, quota errors happen in <5% of the cases and we wouldn't want to have the operator refuse to concilliate and break if it can't get events for whatever reason. It'd still do its thing in the >95% other cases. hence just logging an error and carrying on.

But maybe I'm missing something?

Reply:
my bad, I thought that `log.error` would also return.
So, unfortunately I don't have any practical suggestion, the only feedback is that it's quite hard to parse so many nested (and thus indented) cases. If you see a way to streamline it would be great, otherwise never mind :)